src/main/java/org/candlepin/model/DeletedConsumerCurator.java

```diff
@@ -117,6 +121,98 @@ public int countByConsumerUuid(String uuid) {
             .intValue();
     }
 
+    @Transactional
+    public int upsertDeletedConsumers(Collection<String> consumerIds) {
```

Comment:
nit: I'm not a huge fan of using the term "upsert" as a method verb. I understand why you chose it (it's fundamentally what the implementation does), but it feels like jargon at a level of abstraction where the details aren't supposed to be surfaced.

I think given the work being done here, this may make sense to merge into the actual deleteConsumers operation from an API perspective. As it is now, proper bulk deletion involves calling this method *before* calling the actual deleteConsumers bulk method. Failing to do one without the other will cause data integrity issues, which implies they're both part of the same logical operation. This method could still be retained as a private implementation detail of the delete operation, but probably shouldn't be surfaced as an independent op.